s3_management/update_dependencies.py

```diff
@@ -6,19 +6,61 @@
 CLIENT = boto3.client("s3")
 BUCKET = S3.Bucket("pytorch")
 
-PACKAGES = [
-    "sympy",
-    "mpmath",
-    "pillow",
-    "networkx",
-    "numpy",
-    "jinja2",
-    "filelock",
-    "fsspec",
-    "nvidia-cudnn-cu11",
-    "nvidia-cudnn-cu12",
-    "typing-extensions",
-]
+PACKAGES_PER_PROJECT = {
+    "torch": [
+        "sympy",
+        "mpmath",
+        "pillow",
+        "networkx",
+        "numpy",
+        "jinja2",
+        "filelock",
+        "fsspec",
+        "nvidia-cudnn-cu11",
+        "nvidia-cudnn-cu12",
+        "typing-extensions",
+    ],
+    "torchtune": [
+        "aiohttp",
+        "aiosignal",
+        "antlr4-python3-runtime",
+        "attrs",
+        "blobfile",
+        "certifi",
+        "charset-normalizer",
+        "datasets",
+        "dill",
+        "frozenlist",
+        "huggingface-hub",
+        "idna",
+        "lxml",
+        "markupsafe",
+        "multidict",
+        "multiprocess",
+        "omegaconf",
+        "pandas",
+        "pyarrow",
+        "pyarrow-hotfix",
+        "pycryptodomex",
+        "python-dateutil",
+        "pytz",
+        "pyyaml",
+        "regex",
+        "requests",
+        "safetensors",
+        "sentencepiece",
+        "six",
+        "tiktoken",
+        "torch",
```

Comment:
Here we probably don't want to include: torch, triton. Not sure about torchao

Reply:
For torchao, do we need to include the packages that it requires? Or is that already uploaded to the index somewhere?

Reply:
Not 100% sure. We would need to see how torchao installs the packages now. With "index-url" or "extra-index-url'